x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -328,63 +518,23 @@ protected DataFrameTransformConfig getConfig() {
             return fieldMappings;
         }
 
+        @Override
+        protected DataFrameTransformProgress getProgress() {
+            return progress;
+        }
+
         @Override
         protected String getJobId() {
             return transformId;
         }
 
         @Override
         public synchronized boolean maybeTriggerAsyncJob(long now) {
```

Comment:
This whole method could probably just go away, but this PR kept growing and I was not sure what other refactoring would be necessary if I removed this method and made the parent method `final`

Reply:
:+1: lets do in a follow-up PR